plugins/ocp/ocp-telemetry-decode.c

```diff
@@ -642,43 +644,60 @@ void parse_time_stamp_event(struct nvme_ocp_telemetry_event_descriptor *pevent_d
 {
 	struct nvme_ocp_time_stamp_dbg_evt_class_format *ptime_stamp_event =
 		(struct nvme_ocp_time_stamp_dbg_evt_class_format *) pevent_specific_data;
-
-	int vu_event_id = (int)ptime_stamp_event->vu_event_identifier;
-
-	unsigned int data_size = ((pevent_descriptor->event_data_size * SIZE_OF_DWORD)-
-					sizeof(struct nvme_ocp_time_stamp_dbg_evt_class_format));
-
-	__u8 *pdata = (__u8 *)ptime_stamp_event +
-					sizeof(struct nvme_ocp_time_stamp_dbg_evt_class_format);
-
+	struct nvme_ocp_common_dbg_evt_class_vu_data *ptime_stamp_event_vu_data = NULL;
+	__le16 vu_event_id = 0;
```

Comment:
While this was previously grabbing the little-endian value directly and casting it as an int, now it's using it directly as a little-endian value everywhere else. Instead this should probably remain an integer but convert the parsed value to host-endianness. 

Note that any changes here should also be made in the other event class functions like PCIe Debug, etc.

It might also be worth double-checking how this value is used in the consuming functions to ensure there aren't any other endianness issues.

Reply:
The base event id is declared as a le16 and passed directly to parsing functions as le16.  It is printed correctly in the the parsed output so I believe the vu_event_id should be declared and passed the same (as a le16).  Therefore, I don't think there's any changes needed here.  It's consistent with how the base event id is declared and parsed.

Reply:
The issue is that those event IDs are being treated as unsigned integers in host-endianness, so on a big-endian system they'd be interpreted and printed incorrectly.

Reply:
Fixed the declaration and conversion of the vu_event_id and event_id variables.